lib/web/integrations_awsra.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
+)
+
+// awsRolesAnywhereConfigureTrustAnchor returns a script that configures AWS IAM Roles Anywhere Integration
+// by creating:
+// - IAM Roles Anywhere Trust Anchor which trusts the Teleport AWS RA CA
+// - Roles Anywhere to Apps sync process:
+//   - IAM Role which can be assumed by the Trust Anchor and allows the APIs required by the sync process
+//   - IAM Roles Anywhere Profile which allows access to the IAM Role above
+//
+// It requires the following query parameters:
+// - integrationName: the name of the AWS IAM Roles Anywhere Integration
+// - trustAnchor: the name of the Trust Anchor to be created
+// - syncRole: the name of the IAM Role to be created
+// - syncProfile: the name of the IAM Roles Anywhere Profile to be created
+func (h *Handler) awsRolesAnywhereConfigureTrustAnchor(w http.ResponseWriter, r *http.Request, p httprouter.Params) (any, error) {
+	ctx := r.Context()
+
+	queryParams := r.URL.Query()
+
+	integrationName := queryParams.Get("integrationName")
+	if integrationName == "" {
+		return nil, trace.BadParameter("missing integrationName param")
+	}
+
+	trustAnchorName := queryParams.Get("trustAnchor")
+	if trustAnchorName == "" {
+		return nil, trace.BadParameter("missing trustAnchor param")
+	}
+	if err := aws.IsValidIAMRolesAnywhereTrustAnchorName(trustAnchorName); err != nil {
+		return nil, trace.BadParameter("invalid trustAnchor %q", trustAnchorName)
+	}
+
+	syncRoleName := queryParams.Get("syncRole")
+	if syncRoleName == "" {
+		return nil, trace.BadParameter("missing syncRole param")
+	}
+	if err := aws.IsValidIAMRoleName(syncRoleName); err != nil {
+		return nil, trace.BadParameter("invalid role %q", syncRoleName)
+	}
+
+	syncProfileName := queryParams.Get("syncProfile")
+	if syncProfileName == "" {
+		return nil, trace.BadParameter("missing syncProfile param")
+	}
+	if err := aws.IsValidIAMRolesAnywhereProfileName(syncProfileName); err != nil {
+		return nil, trace.BadParameter("invalid syncProfile %q", syncProfileName)
+	}
+
+	clusterName, err := h.GetProxyClient().GetDomainName(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// Ensure the IntegrationName is valid.
+	_, err = h.GetProxyClient().GetIntegration(ctx, integrationName)
+	// NotFound error is ignored to prevent disclosure of whether the integration exists in a public/no-auth endpoint.
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+
+	authorities, err := client.ExportAllAuthorities(
+		ctx,
+		h.GetProxyClient(),
+		client.ExportAuthoritiesRequest{
+			AuthType: string(types.AWSRACA),
+		},
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if len(authorities) == 0 {
+		return nil, trace.NotFound("no AWS IAM Roles Anywhere CA found")
+	}
+
+	awsRACACertB64 := base64.RawStdEncoding.EncodeToString(authorities[0].Data)
```

Comment:
What if there's more than one authority?

Reply:
Changed to concatenate all the authorities.